lib/src/html/html_generator_instance.dart

```diff
@@ -46,16 +47,48 @@ class HtmlGeneratorInstance {
     }
   }
 
-  void _generateSearchIndex() {
+  _generateCategoryJson() {
+    var encoder = new JsonEncoder.withIndent('  ');
+    final List<Map> indexItems = _categorizationItems.map((Categorization e) {
+      Map data = {
+        'name': e.name,
+        'qualifiedName': e.fullyQualifiedName,
+        'href': e.href,
+        'type': e.kind,
+      };
+
+      if (e.hasCategoryNames) data['categories'] = e.categoryNames;
+      if (e.hasSubCategoryNames) data['subcategories'] = e.subCategoryNames;
+      if (e.hasImage) data['image'] = e.image;
+      if (e.hasSamples) data['samples'] = e.samples;
+      return data;
+    }).toList();
+
+    indexItems.sort((a, b) {
+      var value = compareNatural(a['qualifiedName'], b['qualifiedName']);
+      if (value == 0) {
+        value = compareNatural(a['type'], b['type']);
+      }
+      return value;
+    });
+
+    String json = encoder.convert(indexItems);
+    _writer(pathLib.join('categories.json'), '${json}\n');
+  }
+
+  List<Categorization> _categorizationItems;
+  _generateSearchIndex() {
```

Comment:
ditto re: the return type (is it common for the dartdoc to not specify return types?)

We may want to enable the lint to detect missing return types.

Reply:
done.